src/network/model/socket.cc

```diff
@@ -68,14 +69,14 @@ Socket::~Socket ()
 }
 
 Ptr<Socket> 
-Socket::CreateSocket (Ptr<Node> node, TypeId tid)
+Socket::CreateSocket (Ptr<Node> node, TypeId tid, TypeId congestionTypeId)
 {
   NS_LOG_FUNCTION (node << tid);
   Ptr<Socket> s;
   NS_ASSERT (node != 0);
-  Ptr<SocketFactory> socketFactory = node->GetObject<SocketFactory> (tid);
+  Ptr<TcpSocketFactoryImpl> socketFactory = node->GetObject<TcpSocketFactoryImpl> (tid);
```

Comment:
Would you please add a comment noting that this means that we're restricted to TCP sockets?